src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -555,7 +555,21 @@ public void AddLowercase(CultureInfo culture)
                     }
                     else
                     {
-                        AddLowercaseRange(range.First, range.Last);
+                        char lower = culture.TextInfo.ToLower(range.First);
+                        char upper = culture.TextInfo.ToLower(range.Last);
+                        if (range.Last - range.First == upper - lower)
+                        {
+                            // Bug fix: https://github.com/dotnet/runtime/issues/36149
```

Comment:
I'd prefer if the comment actually described the condition here rather than linking off to something else.